### DISCUSSION TITLE
Old Patchmanager 4 branch

### DISCUSSION THREAD

**Discussion by Olf0** [sailfishos-patches/patchmanager#151] [General]
I always wondered what Andrey planned in detail for Patchmanager 4, looking at the [`pm4` branch](https://github.com/sailfishos-patches/patchmanager/commits/pm4):
- Structurally it contains a [single big commit](https://github.com/sailfishos-patches/patchmanager/commit/aea5f4ccc0544baf6dde46d10f8704a2ecd936ce) on top of Patchmanager 3 in February 2019.
  This is a bit unfortunate (technically no "cherry picking" possible, a little hard to review / analyse), but somewhat understandable, because ...
- Its changes seem to be basically clean-up work, which apparently were done locally in one go.

@CODeRUS, can you briefly describe what your intentions / goals were for PM4, i.e., what you wanted to achieve / change compared to PM3?
Do you think there may be specific changes, which are still worth being analysed and considered?  Or do you advise to ignore that code / branch altogether?

**Comment by b100dian** [comment]
I got a few lines chatting with Andrey - basically asking the same question, what's the gist/idea behind this branch.
He said that the reason was he was planning a full rewrite, since the
> current architecture is painful. based on old code based on old code. pathmanagerobject superclass is bad, hard to maintain

However these were only plans - it sounded to me like a first commit to make a "clean break" from the pm3 stuff.

While it sounds like a good plan to have something cleanly redone, it doesn't seem to me like an architectural change the magnitude pm2 vs pm3 is.

I think we can archive that branch for posterity:)

**Reply by Olf0** [reply]
Thank you for (obtaining the) information!
